src/Network/NWPathMonitor.cs

```diff
@@ -54,5 +54,78 @@ public void SetQueue (DispatchQueue queue)
 				throw new ArgumentNullException (nameof (queue));
 			nw_path_monitor_set_queue (GetCheckedHandle (), queue.Handle);
 		}
+
+		delegate void nw_path_monitor_update_handler_t (IntPtr block, IntPtr path);
+		static nw_path_monitor_update_handler_t static_UpdateSnapshot = TrampolineUpdatedSnapshot;
+
+		[MonoPInvokeCallback (typeof (nw_path_monitor_update_handler_t))]
+		static void TrampolineUpdatedSnapshot (IntPtr block, IntPtr path)
+		{
+			var del = BlockLiteral.GetTarget<Action<NWPath>> (block);
+			if (del != null) {
+				var nwPath = new NWPath (path, owns: false);
+				del (nwPath);
+			}
+		}
+
+		[DllImport (Constants.NetworkLibrary)]
+		static extern unsafe void nw_path_monitor_set_update_handler (IntPtr handle, void *callback);
+
+		[BindingImpl (BindingImplOptions.Optimizable)]
+		public void SetUpdatedSnapshotHandler (Action<NWPath> callback)
```

Comment:
why `Snapshot` in the name ?

Reply:
From the documentation of the headers

`
+/*!
+ * @typedef nw_path_monitor_update_handler_t
+ * @abstract
+ *		A block handler to be called whenever a path monitor has a new snapshot.
+ *		This will be called one or more times after calling nw_path_monitor_start.
+ */
+typedef void (^nw_path_monitor_update_handler_t) (nw_path_t path);
+
+/*!
+ * @function nw_path_monitor_set_update_handler
+ * @abstract
+ *		Sets the client update handler. This block will be called with the
+ *		current path when start is called and any time the path changes.
+ *
+ * @param monitor
+ *		The path monitor object.
+ *
+ * @param update_handler
+ *		The block to call when the path changes.
+ */
+API_AVAILABLE(macos(10.14), ios(12.0), watchos(5.0), tvos(12.0))
+void
+nw_path_monitor_set_update_handler(nw_path_monitor_t monitor,
+								   nw_path_monitor_update_handler_t update_handler);
`